liquibase-couchbase/src/test/java/liquibase/ext/couchbase/provider/ContextServiceProviderTest.java

```diff
@@ -0,0 +1,301 @@
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class ContextServiceProviderTest {
+
+    private final CouchbaseLiquibaseDatabase couchbaseLiquibaseDatabase = mock(CouchbaseLiquibaseDatabase.class);
+    private final CouchbaseConnection couchbaseConnection = mock(CouchbaseConnection.class);
+    private final Cluster cluster = mock(Cluster.class);
+    private final Bucket bucket = mock(Bucket.class);
+    private final Scope scope = mock(Scope.class);
+    private final Collection collection = mock(Collection.class);
+    private final BucketManager bucketManager = mock(BucketManager.class);
+    private final CollectionManager collectionManager = mock(CollectionManager.class);
+    private final QueryIndexManager queryIndexManager = mock(QueryIndexManager.class);
+
+    @BeforeEach
+    void setUp() {
+        reset(couchbaseLiquibaseDatabase, couchbaseConnection, cluster, bucket, scope, cluster, bucketManager,
+                collectionManager, queryIndexManager);
+        when(couchbaseLiquibaseDatabase.getConnection()).thenReturn(couchbaseConnection);
+        when(couchbaseConnection.getCluster()).thenReturn(cluster);
+        when(bucket.collections()).thenReturn(collectionManager);
+        when(cluster.queryIndexes()).thenReturn(queryIndexManager);
+        when(collection.bucketName()).thenReturn(SERVICE_BUCKET_NAME);
+        when(collection.scopeName()).thenReturn(DEFAULT_SERVICE_SCOPE);
+    }
+
+    @Test
+    void Should_return_scope_of_collection() {
+        ContextServiceProvider contextServiceProvider = new ContextServiceProvider(couchbaseLiquibaseDatabase);
+
+        String collectionName = "collection";
+
+        List<ScopeSpec> scopeSpecList = createScopeSpecs(DEFAULT_SERVICE_SCOPE);
+        scopeSpecList.forEach(it -> {
+            String scopeName = it.name();
+            Set<CollectionSpec> collectionSpecSet = createCollectionSpecs(scopeName, collectionName);
+            when(it.collections()).thenReturn(collectionSpecSet);
+        });
+        List<QueryIndex> queryIndexList = createQueryIndexes(collectionName);
+        when(couchbaseConnection.getDatabase()).thenReturn(bucket);
+        when(bucket.name()).thenReturn(SERVICE_BUCKET_NAME);
+        when(cluster.bucket(SERVICE_BUCKET_NAME)).thenReturn(bucket);
+        when(cluster.buckets()).thenReturn(bucketManager);
+        when(bucket.scope(DEFAULT_SERVICE_SCOPE)).thenReturn(scope);
+        when(scope.collection(collectionName)).thenReturn(collection);
+        when(collectionManager.getAllScopes()).thenReturn(scopeSpecList);
+        when(queryIndexManager.getAllIndexes(eq(SERVICE_BUCKET_NAME), any())).thenReturn(queryIndexList);
+
+        assertThat(contextServiceProvider.getScopeOfCollection(collectionName)).isEqualTo(scope);
```

Comment:
Pls extract the method call to a variable like `result` to simplify assertion check

Reply:
Ok, did it